src/test/java/formflow/library/framework/BeforeDisplayTest.java

```diff
@@ -0,0 +1,58 @@
+package formflow.library.framework;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.when;
+
+import formflow.library.ScreenController;
+import formflow.library.data.Submission;
+import formflow.library.data.SubmissionRepositoryService;
+import formflow.library.utilities.AbstractMockMvcTest;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.test.web.servlet.setup.MockMvcBuilders;
+
+@SpringBootTest(properties = {"form-flow.path=flows-config/test-before-display.yaml"})
+public class BeforeDisplayTest extends AbstractMockMvcTest {
+
+  Submission submission;
+
+  private MockMvc mockMvc;
+
+  @MockBean
+  private SubmissionRepositoryService submissionRepositoryService;
+
+  @Autowired
+  private ScreenController screenController;
+
+  @BeforeEach
+  public void setUp() throws Exception {
+    mockMvc = MockMvcBuilders.standaloneSetup(screenController).build();
+    submission = Submission.builder().id(1L).inputData(new HashMap<>()).build();
+
+    super.setUp();
+    when(submissionRepositoryService.findOrCreate(any())).thenReturn(submission);
+    when(submissionRepositoryService.findById(any())).thenReturn(Optional.of(submission));
+  }
+
+  @Test
+  void shouldSaveFormattedDate() throws Exception {
+    postExpectingSuccess("inputs",
+        Map.of("ssnInput", List.of("111-00-1234")));
+    assertThat(submission.getInputData().get("ssnInputEncrypted")).isEqualTo("BBB-AA-BCDE");
+    Assertions.assertNull(submission.getInputData().get("ssnInput"));
+
+    getPageExpectingSuccess("inputs");
+    assertThat(submission.getInputData().get("ssnInput")).isEqualTo("111-00-1234");
+  }
```

Comment:
It would be worth testing here to make sure the page's data doesn't have "ssnInputEncrypted" and that submission data on line 51 doesn't have "ssnInput".

Reply:
Er- how do I get the page's data here?